org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/alloy4/Util.java

```diff
@@ -302,6 +302,22 @@ public static String readAll(String filename) throws FileNotFoundException, IOEx
         return convertLineBreak(ans);
     }
 
+    /** Returns the modified date of the given file
+     * (If filename begins with Util.jarPrefix() returns 0) 
+     */
+    public static long getModifiedDate(String filename){
+        String JAR = jarPrefix();
+        boolean fromJar = false;
+        if (filename.startsWith(JAR)) {
+            fromJar = true;
+        }
+        if (! fromJar) {
```

Comment:
one more nit: in the rest of the codebase, there is usually no space betwee `!` and expression.